### PR TITLE
Move to intra-doc links in collections/btree/map.rs and collections/linked_list.rs

### PR DESCRIPTION
**Pull request by denisvasilik**
Helps with #75080.

@rustbot modify labels: T-doc, A-intra-doc-links

### CODE REVIEW

**Comment by denisvasilik** [comment]
`std/collections/struct.BTreeMap.html:265: broken link fragment `#insert-and-complex-keys` pointing to `std/index.html``

I'm on it.

File: library/alloc/src/collections/btree/map.rs [hunk 1]
```diff
@@ -815,7 +811,7 @@ impl<K: Ord, V> BTreeMap<K, V> {
     /// types that can be `==` without being identical. See the [module-level
     /// documentation] for more.
     ///
-    /// [module-level documentation]: index.html#insert-and-complex-keys
+    /// [module-level documentation]: crate#insert-and-complex-keys
```
Comment (jyn514) [inline library/alloc/src/collections/btree/map.rs:814]:
```suggestion
    /// [module-level documentation]: self#insert-and-complex-keys
```

**Review by jyn514 — COMMENTED** [review]
r=me with CI passing

**Comment by denisvasilik** [comment]
I think `self#insert-and-complex-keys` should be the correct link, but there's again the problem with private items. So I'm going to revert that change.

```
814 |     /// [module-level documentation]: self#insert-and-complex-keys
    |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ this item is private
```

File: library/alloc/src/collections/btree/map.rs (hunk 1 above)
Comment (denisvasilik) [reply library/alloc/src/collections/btree/map.rs:814]: That should be it, but I get the following error. I think we had that same problem with another link to a module level doc.

```
814 |     /// [module-level documentation]: self#insert-and-complex-keys
    |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ this item is private
```

Comment (jyn514) [reply library/alloc/src/collections/btree/map.rs]: Hmm, maybe this is meant to be shown on the re-export? Try `super#` and see if that helps/generates the right link.

Comment (denisvasilik) [reply library/alloc/src/collections/btree/map.rs]: I tried your suggestion, but it didn't help. I got the same error.
```
814 |     /// [module-level documentation]: super#insert-and-complex-keys
    |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ this item is private
```

Comment (jyn514) [reply library/alloc/src/collections/btree/map.rs]: Ok, I took the time to look it up - it's linking to https://doc.rust-lang.org/std/collections/index.html#insert-and-complex-keys. Wow, that is not clear at all from the current text.

Try `[module-level documentation]: crate::collections#insert-and-complex-keys`.

Comment (denisvasilik) [reply library/alloc/src/collections/btree/map.rs]: This looks good, thanks for helping me out.

**Comment by jyn514** [comment]
r=me with commits squashed, thanks for sticking with this! Let me know if you have trouble with git and I can help.

**Comment by jyn514** [comment]
@bors r+ rollup

Next time I'll @bors delegate denisvasilik, sorry about that

**Pull request merged**